File: apis/backend-for-frontend/src/constants/BaseError.ts
export default class BaseError extends Error {
  public readonly name: string;
  public readonly statusCode: number;
  public readonly isOperational: boolean;

  constructor({ name, statusCode, isOperational, message }: BaseError) {
    super(message);
    Object.setPrototypeOf(this, new.target.prototype);

    this.name = name;
    this.statusCode = statusCode;
    this.isOperational = isOperational;

    Error.captureStackTrace(this);
  }
}

File: apis/playlists-service/internal/constants/baseError.go
package constants

import (
	"encoding/json"
	"fmt"
)

type BaseError struct {
	Name          string
	Message       string
	StatusCode    int32
	IsOperational bool
	Details       interface{}
}

func (e *BaseError) Error() string {
	var p []byte
	p, err := json.MarshalIndent(e, "", "\t")
	if err != nil {
		return fmt.Sprint(err)
	}
	return fmt.Sprintf("%s \n", p)
}
